### ISSUE TITLE
Fix memleaks identified by pio_decomp_fillval test

### ISSUE THREAD

**Issue by jayeshkrishna** [NCAR/ParallelIO#305]
There are some memory leaks identified by the pio_decomp_fillval test with valgrind.

**Comment by jayeshkrishna** [comment]
Couple of cases that might need to go here,
```
diff --git a/src/clib/pioc_support.c b/src/clib/pioc_support.c
index 4ba7b77..b031c94 100644
--- a/src/clib/pioc_support.c
+++ b/src/clib/pioc_support.c
@@ -568,6 +568,9 @@ int PIOc_freedecomp(int iosysid, int ioid)
     if (!(iodesc = pio_get_iodesc_from_id(ioid)))
         return PIO_EBADID;
 
+    if (iodesc->gsize)
+        brel(iodesc->gsize);
+
     if (iodesc->rfrom)
         brel(iodesc->rfrom);
     
@@ -604,6 +607,9 @@ int PIOc_freedecomp(int iosysid, int ioid)
     if (iodesc->firstregion)
         free_region_list(iodesc->firstregion);
 
+    if (iodesc->fillregion)
+        brel(iodesc->fillregion);
+
     if (iodesc->rearranger == PIO_REARR_SUBSET)
         MPI_Comm_free(&(iodesc->subset_comm));
```

**Comment by jayeshkrishna** [comment]
There are also some memory allocations (that are leaked) that happen only when fillvalues are used.

**Comment by jayeshkrishna** [comment]
@dqwu : Please feel free to create multiple PRs related to this issue if the code that needs to be modified is distributed across the code (or if non-trivial changes are required, separate it out into a different PR). As @edhartnett noted before it would also help the review process to have an issue for non-trivial changes.

**Comment by dqwu** [comment]
The added code brel(iodesc->fillregion) should be free_region_list(iodesc->fillregion).

**Comment by dqwu** [comment]
For pio_decomp_fillval test, there are still some leaks reported by Valgrind after PR #328 is merged to the develop branch (previous: "5,172 bytes in 372 blocks", current: "660 bytes in 60 blocks"). However, the remaining leaks are from pnetcdf 1.6.1, which should have been fixed in pnetcdf 1.7.0. We will run Valgrind check on PIO built with pnetcdf 1.7.0 or higher versions.
